### PR TITLE
[CDEC-505] Move Pos.Core.Update to Pos.Chain.Update

### PR DESCRIPTION
**Pull request by ruhatch**
## Description

Last step of CDEC-505, moving `Pos.Core.Update` module set to the `chain` package

## Linked issue

[CDEC-505](https://iohk.myjetbrains.com/youtrack/issue/CDEC-505)

## Type of change

- [~] 🐞 Bug fix (non-breaking change which fixes an issue)
- [~] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [~] 🔨 New or improved tests for existing code
- [~] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [~] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [~] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: chain/test/Test/Pos/Chain/Update/Gen.hs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    upDataList <- Gen.list (Range.singleton hMapSize) genUpdateData
+    pure $ HM.fromList $ zip sysTagList upDataList
+
+genUpdateVote :: ProtocolMagic -> Gen UpdateVote
+genUpdateVote pm = mkUpdateVote pm <$> genSecretKey <*> genUpId pm <*> Gen.bool
+        -- <$> genSecretKey
+        -- <*> genUpId
+        -- <*> Gen.bool
+        -- <*> genSignature ((,) <$> genUpId <*> Gen.bool)
+-- genUpdateVote pm = mkUpdateVote pm <$> arbitrary <*> arbitrary <*> arbitrary
```
Comment (parsonsmatt) [inline chain/test/Test/Pos/Chain/Update/Gen.hs]: Dead code?

File: chain/test/Test/Pos/Chain/Update/Gen.hs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+genSoftwareVersion =
+    SoftwareVersion
+        <$> genApplicationName
+        <*> Gen.word32 Range.constantBounded
+
+genSystemTag :: Gen SystemTag
+genSystemTag = SystemTag <$> Gen.text (Range.constant 0 10) Gen.alphaNum
+
+genUpAttributes :: Gen UpAttributes
+genUpAttributes = pure $ mkAttributes ()
```
[the remark this answers is not in the document]
Comment (ruhatch) [reply chain/test/Test/Pos/Chain/Update/Gen.hs]: So far yes - the `Attributes` type is for extending records that are stored in the blockchain

File: chain/test/Test/Pos/Chain/Update/Arbitrary.hs [hunk 3]
```diff
@@ -1,9 +1,124 @@
...
+    shrink = genericShrink
+
+instance Arbitrary BlockVersionModifier where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+instance Arbitrary SystemTag where
+    arbitrary =
+        oneof . map (pure . SystemTag) $
+        [os <> arch | os <- ["win", "linux", "mac"], arch <- ["32", "64"]]
```
[the remark this answers is not in the document]
Comment (ruhatch) [reply chain/test/Test/Pos/Chain/Update/Arbitrary.hs]: I agree, or `(<>) <$> ["win", "linux", "mac"] <*> ["32", "64"]`

File: chain/test/Test/Pos/Chain/Update/Gen.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (parsonsmatt) [reply chain/test/Test/Pos/Chain/Update/Gen.hs]: Neat

**Pull request merged**